src/ucp/proto/proto_select.inl

```diff
@@ -62,7 +62,8 @@ ucp_proto_select_op_attr_pack(uint32_t op_attr_mask)
     UCS_STATIC_ASSERT(
             (UCP_PROTO_SELECT_OP_ATTR_MASK / UCP_PROTO_SELECT_OP_ATTR_BASE) <
             UCP_PROTO_SELECT_OP_FLAGS_BASE);
-    return op_attr_mask / UCP_PROTO_SELECT_OP_ATTR_BASE;
+    return (op_attr_mask & UCP_PROTO_SELECT_OP_ATTR_MASK) /
```

Comment:
why needed - can't we just ignore unused flags?

Reply:
This is a fix for fast proto selection: https://github.com/openucx/ucx/blob/fe89ac6de9ded33b1a38b12aacdd6dbcf0ab4a2f/src/ucp/proto/proto_select.inl#L109

Reply:
you mean we have cache miss because of irrelevant flags in op_attr_mask? so it's a performance issue?
can you pls update PR desc accordingly?

Reply:
yes.
e.g. we send with `UCP_OP_ATTR_FLAG_NO_IMM_CMPL`, then `select_param` will reflect the flag. And protocol selection will go through the slow path.
Along with that we print the same configuration twice in case of `UCX_PROTO_INFO=y`.
Thus, it is a performance issue, and duplication it protocol information output.

Reply:
P.s. With these changes there is no need to apply mask in `ucp_proto_select_op_attr_unpack`.
I will update the PR.